common/persistence/persistenceInterface.go

```diff
@@ -22,6 +22,8 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+//go:generate mockgen -copyright_file ../../LICENSE -package mock -destination mock/store_mock.go go.temporal.io/server/common/persistence ShardStore,TaskStore,HistoryStore,ExecutionStore,MetadataStore,ClusterMetadataStore,VisibilityStore,Queue
```

Comment:
Why not to use `source` mode? (as we do everywhere else).

Reply:
I tried that first, and it did not work because the file uses Closeable as embedded interface which is not defined in this file. Not sure if there is a better way to do so.
2021/06/15 19:13:47 Loading input failed: persistenceInterface.go:54:3: unknown embedded interface Closeable
common/persistence/persistenceInterface.go:25: running "mockgen": exit status 1
make: *** [go-generate] Error 1

Reply:
15m of googling, turns out i need to use -aux_files to supply source code for embedded interfaces.

Reply:
We even have it in few places already.